packages/store-sdk/src/storage/useLocalStorage.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Safe localstorage interface. These try..catch are usefull because
+ * some browsers may block accesss to these APIs due to security policies
+ *
+ * Also, the local storage value is lazy-loaded to avoid hydration mismatch
+ * between server/browser. When state is 'hydrated', the in the heap is the
+ * same as the value in local storage
+ */
+import { useState, useEffect, useCallback } from 'react'
+
+const getItem = <T>(key: string) => {
```

Comment:
I think we should consider using [localForage](https://github.com/localForage/localForage) for this. It's not a heavy library (but certainly heavier than this ofc) and it brings numerous advantages over using only localStorage directly. I think it's way more scalable to go with localForage.

Reply:
I'm not sure. According to bundlephobia this lib has 8.5 min+gzip, which is huge, specially comparing to other deps, like swr that has 5 min+gzip and handles the whole IO or `react-use-cart` that is 1.5 min+gzip

Reply:
What do you think on leaving this localStorage iimplementation and if we see this doesn't scale well we use a different approach

Reply:
Makes sense. Then I propose for us to migrate to IndexedDB, since it doesn't block the main thread as LocalStorage does. We also gain more freedom regarding storage quotas.

I have found [this good read](https://web.dev/storage-for-the-web/).
> LocalStorage should be avoided because it is synchronous and will block the main thread. It is limited to about 5MB and can contain only strings. LocalStorage is not accessible from web workers or service workers.

Reply:
> What do you think on leaving this localStorage iimplementation and if we see this doesn't scale well we use a different approach

I only worry about having to release a major for store-sdk only to upgrade it to indexeddb. I think that for 99% of use-cases, localStorage will be enough, but we would have to break this 99% because of the 1% that isn't contemplated. I don't think transferring data between systems would be a trivial task.

indexeddb is by far more complicated to use, but I think we can make an opportunity out of this. Maybe create a super simple to manage wrapper around indexeddb that's also super lightweight.

I'm okay with using localstorage as long we don't go stable without changing it to indexeddb. For me, that's a must.